src/main/scala/tech/sourced/gemini/Gemini.scala

```diff
@@ -108,6 +118,21 @@ class Gemini(session: SparkSession, log: Slf4jLogger, keyspace: String = Gemini.
     resp.uast
   }
 
+  def extractFeatures(client: FeatureExtractor, uast: Node): Iterable[Feature] = {
+    val idRequest = IdentifiersRequest(uast=Some(uast), docfreqThreshold=5)
+    val litRequest = LiteralsRequest(uast=Some(uast), docfreqThreshold=5)
+    val uast2seqRequest = Uast2seqRequest(uast=Some(uast), docfreqThreshold=5)
+    client.identifiers(idRequest)
+
+    val features = for {
+      idResponse <- client.identifiers(idRequest)
+      litResponse <- client.literals(litRequest)
+      uast2seqResponse <- client.uast2Seq(uast2seqRequest)
+    } yield idResponse.features ++ litResponse.features ++ uast2seqResponse.features
+
+    Await.result(features, Duration(30, SECONDS))
```

Comment:
it would print `log.error` for now. I decided to keep it as `log` message instead of a crush because duplication functionality is still usable without similarity. But I see other possibilities like make it mandatory or configure by a flag. I'm very welcome to any input about it (imo, the best way would be to discuss and create an issue)

But for current integration tests, this code is never executed because we test with `LICENCE` file. It is a text, not a code. So UAST extractor returns empty UAST.

I don't see for now how to write CI tests for query, I tested it only locally. CI is checking that the code is compilable only. I think writing unit tests for 10 lines of code that doesn't do anything but calls api isn't beneficial. The API client is already covered by unit tests.

For now, my plan is:
- merge "UAST extraction" (locally tested)
- merge "extract Features" (locally tested)
- implement "implement actual query logic" and make `findSimilarForFile` actually return results. Write CI tests to check those results. (not as a part of integration testing but as unit tests with mocked dependencies)
- implement reporting (with unit tests) and implement full integration test.

Unit tests would require meaningful fixtures and without "actual query logic" I can't create such.